libbeat/common/field.go

```diff
@@ -304,7 +344,7 @@ func ConcatFields(a, b Fields) (Fields, error) {
 
 	// check for duplicates
 	for _, k := range b.GetKeys() {
-		if a.HasNode(k) {
+		if !a.CanConcat(k) {
```

Comment:
This was the only caller of `HasNode`, can that be removed now?   Can this implementation replace the previous one instead of introducing a new function?

Reply:
I didn't remove it as it is an exported method, and we should backport the bugfix to 6.x (best 6.6).